src/nlp.jl

```diff
@@ -498,32 +498,21 @@ function MathProgBase.constr_expr(d::JuMPNLPEvaluator,i::Integer)
     end
 end
 
-function _buildInternalModel_nlp(m::Model, traits)
-
-    @assert isempty(m.sdpconstr) && isempty(m.socconstr)
+getConstraintBounds(m::Model) = getConstraintBounds(m, ProblemTraits(m))
 
-    linobj, linrowlb, linrowub = prepProblemBounds(m)
+function getConstraintBounds(m::Model,traits::ProblemTraits)
```

Comment:
The logic of flattening out constraints into linear then quadratic then nonlinear has so far been constrained to ``nlp.jl``, I think.